src/common/fidgets/FidgetWrapper.tsx

```diff
@@ -13,12 +13,16 @@ import {
 import { FaGear } from "react-icons/fa6";
 import FidgetSettingsPopover from "@/common/fidgets/FidgetSettingsPopover";
 import { reduce } from "lodash";
+import FidgetSettingsEditor from "../components/organisms/FidgetSettingsEditor";
 
 export type FidgetWrapperProps = {
   fidget: React.FC<FidgetArgs>;
   config: FidgetDetails;
   context?: FidgetRenderContext;
   saveConfig: (conf: FidgetConfig) => Promise<void>;
+  setcurrentFidgetSettings: (currentFidgetSettings: React.JSX.Element) => void;
```

Comment:
These three should be rolled up into `onEnterEditMode` + `onExitEditMode` functions that the parent can control how they works. The wrapper doesn't need to have a concept of the selection process, only that it is supposed to call `onEnterEditMode` with a bunch of information (fidget ID + Editor Component) each time if enters edit mode and call `onExitEditMode` each time it leaves